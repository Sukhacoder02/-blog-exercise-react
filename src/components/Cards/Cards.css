.cards {
  width: 80%;
  padding: 0 10%;
  background-color: white;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  gap: 20px 20px;
}

.blogDataError {
  display: flex;
  align-items: center;
  justify-content: center;
}
