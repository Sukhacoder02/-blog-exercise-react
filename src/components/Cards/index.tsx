import * as React from 'react';
import Card from './../Card';
import './Cards.css';
import postsJson from '../../mockData/index.json';

const Cards: React.FunctionComponent = (): JSX.Element => {
  const cards = postsJson.map((post, index) => {
    return (
      <Card
        key={index}
        date={post.date}
        readingTime={post.readingTime}
        title={post.title}
        description={post.description}
        claps={post.claps}
        image={post.image}
        liked={post.liked}
      />
    );
  });
  return <div className="cards">{cards}</div>;
};
export default Cards;
