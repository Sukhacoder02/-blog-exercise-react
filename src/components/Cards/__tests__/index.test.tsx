import * as React from 'react';
import { render } from '@testing-library/react';
import Cards from '..';

describe('Component Cards', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Cards />);
    expect(asFragment()).toMatchSnapshot();
  });
});
