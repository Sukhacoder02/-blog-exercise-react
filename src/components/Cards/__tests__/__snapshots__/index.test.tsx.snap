// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Cards renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="cards"
  >
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="Image"
          class="cardImage"
          src="/Images/abstract.png"
        />
      </div>
      <div
        class="belowImage"
      >
        <div
          class="textBelowImage"
        >
          <p>
            2nd January, 2018
          </p>
          <p>
            2 mins
          </p>
        </div>
        <div
          class="title"
        >
          <h3>
            The future of abstract art and the culture ...
          </h3>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="hands"
          >
            <button
              class="claps"
              data-testid="testClapButton"
            >
              <img
                alt="image"
                src="/Icons/clapping.svg"
              />
            </button>
            <p
              data-testid="claps"
            >
              10
            </p>
          </div>
          <div
            class="heart"
          >
            <button
              data-testid="testHeartButton"
            >
              <img
                alt="image"
                src="/Icons/heart-black.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="Image"
          class="cardImage"
          src="/Images/outdoor.png"
        />
      </div>
      <div
        class="belowImage"
      >
        <div
          class="textBelowImage"
        >
          <p>
            31st January, 2018
          </p>
          <p>
            1 min
          </p>
        </div>
        <div
          class="title"
        >
          <h3>
            How painting outdoors can transform your 
          </h3>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="hands"
          >
            <button
              class="claps"
              data-testid="testClapButton"
            >
              <img
                alt="image"
                src="/Icons/clapping.svg"
              />
            </button>
            <p
              data-testid="claps"
            >
              1
            </p>
          </div>
          <div
            class="heart"
          >
            <button
              data-testid="testHeartButton"
            >
              <img
                alt="image"
                src="/Icons/heart-red.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="Image"
          class="cardImage"
          src="/Images/young-painters.png"
        />
      </div>
      <div
        class="belowImage"
      >
        <div
          class="textBelowImage"
        >
          <p>
            20th March, 2018
          </p>
          <p>
            2 mins
          </p>
        </div>
        <div
          class="title"
        >
          <h3>
            10 young painters you need to know
          </h3>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="hands"
          >
            <button
              class="claps"
              data-testid="testClapButton"
            >
              <img
                alt="image"
                src="/Icons/clapping.svg"
              />
            </button>
            <p
              data-testid="claps"
            >
              15
            </p>
          </div>
          <div
            class="heart"
          >
            <button
              data-testid="testHeartButton"
            >
              <img
                alt="image"
                src="/Icons/heart-black.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="Image"
          class="cardImage"
          src="/Images/perfect-art.png"
        />
      </div>
      <div
        class="belowImage"
      >
        <div
          class="textBelowImage"
        >
          <p>
            22nd June, 2018
          </p>
          <p>
            5 mins
          </p>
        </div>
        <div
          class="title"
        >
          <h3>
            Tips for creating the perfect art
          </h3>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="hands"
          >
            <button
              class="claps"
              data-testid="testClapButton"
            >
              <img
                alt="image"
                src="/Icons/clapping.svg"
              />
            </button>
            <p
              data-testid="claps"
            >
              11
            </p>
          </div>
          <div
            class="heart"
          >
            <button
              data-testid="testHeartButton"
            >
              <img
                alt="image"
                src="/Icons/heart-black.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="Image"
          class="cardImage"
          src="/Images/inspired.png"
        />
      </div>
      <div
        class="belowImage"
      >
        <div
          class="textBelowImage"
        >
          <p>
            4th February, 2019
          </p>
          <p>
            9 mins
          </p>
        </div>
        <div
          class="title"
        >
          <h3>
            Stay inspired: top artists blog ...
          </h3>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="hands"
          >
            <button
              class="claps"
              data-testid="testClapButton"
            >
              <img
                alt="image"
                src="/Icons/clapping.svg"
              />
            </button>
            <p
              data-testid="claps"
            >
              45
            </p>
          </div>
          <div
            class="heart"
          >
            <button
              data-testid="testHeartButton"
            >
              <img
                alt="image"
                src="/Icons/heart-red.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="img"
      >
        <img
          alt="Image"
          class="cardImage"
          src="/Images/reason.png"
        />
      </div>
      <div
        class="belowImage"
      >
        <div
          class="textBelowImage"
        >
          <p>
            10th January, 2019
          </p>
          <p>
            2 mins
          </p>
        </div>
        <div
          class="title"
        >
          <h3>
            5 reasons why everyone should try an...
          </h3>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="hands"
          >
            <button
              class="claps"
              data-testid="testClapButton"
            >
              <img
                alt="image"
                src="/Icons/clapping.svg"
              />
            </button>
            <p
              data-testid="claps"
            >
              11
            </p>
          </div>
          <div
            class="heart"
          >
            <button
              data-testid="testHeartButton"
            >
              <img
                alt="image"
                src="/Icons/heart-black.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
